### ISSUE TITLE
Turn off pretty errors

### ISSUE THREAD

**Issue by rizowski** [prisma/photonjs#195]
When creating the generated client, is there a way for it to return regularly formatted errors without the line numbers or coloring? Or are there properties on the error object where I can look for codes or something that doesn't include all the unicode characters? I've been looking around trying to find what might include this information but it isn't very apparent.

**On create failure console example:**
```
PhotonError: 
Invalid `photon.users.create()` invocation in /Users/rizowski/git/work/auth/server/graphql/controllers/users.js:7:22

{
  username: 'test',
  ~~~~~~~~
  password: 'secret',
  ~~~~~~~~
  email: 'test@test.com',
  ~~~~~
+ data: {
+   id?: ID,
+   username: String,
+   email: String,
+   password: String,
+   created?: DateTime,
+   updated?: DateTime,
+   role?: EMPLOYEE | TRAINER | CLIENT | PARTNER,
+   resetPassToken: String,
+   resetPassExpiration: DateTime
+ }
}

Unknown arg `username` in username for type User. Did you mean `select`?
Unknown arg `password` in password for type User. Did you mean `data`?
Unknown arg `email` in email for type User. Did you mean `data`?
Argument data for data is missing.

Note: Lines with + are required

    at Document.validate (/Users/rizowski/git/workout/auth/node_modules/@generated/photon/runtime/index.js:1718:23)
    at UserClient.get _document [as _document] (/Users/rizowski/git/workout/auth/node_modules/@generated/photon/index.js:289:22)
    at UserClient.then (/Users/rizowski/git/workout/auth/node_modules/@generated/photon/index.js:310:63)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
```

**String when serialized:**
```
\nInvalid `photon.users.create()` invocation in /Users/rizowski/git/workout/auth/server/graphql/controllers/users.js:7:22\n\n{\n  username: 'test',\n  ~~~~~~~~\n  password: 'secret',\n  ~~~~~~~~\n  email: 'test@test.com',\n  ~~~~~\n+ data: {\n+   id?: ID,\n+   username: String,\n+   email: String,\n+   password: String,\n+   created?: DateTime,\n+   updated?: DateTime,\n+   role?: EMPLOYEE | TRAINER | CLIENT | PARTNER,\n+   resetPassToken: String,\n+   resetPassExpiration: DateTime\n+ }\n}\n\nUnknown arg `username` in username for type User. Did you mean `select`?\nUnknown arg `password` in password for type User. Did you mean `data`?\nUnknown arg `email` in email for type User. Did you mean `data`?\nArgument data for data is missing.\n\nNote: Lines with + are required\n
```

Again I don't think these kinds of errors are bad when you are developing but having the ability to make them less verbose would be nice for production environments.

Suggestions:

I personally like having error codes come back on my error objects. It makes it nice for trying to decipher what is wrong programmatically a lot simpler. (As long as they don't change)

The pretty text could be fine if it is attached to the error object but maybe not as the main error message. Ideally it can be turned off or changed to have multiple properties on the error that describe the same information.

**Comment by pantharshit00** [comment]
Closing as a duplicate of https://github.com/prisma/prisma2/issues/313

**Comment by pantharshit00** [comment]
I guess we can also leave this open here for photon js. 

The other issue only concerns the CLI.

**Comment by schickling** [comment]
@rizowski out of curiosity: Could you explain your use case a bit more?

**Comment by rizowski** [comment]
So in the case of `db.users.findOne()`

If the record isn't found it throws an exception. That particular exception is pretty formatted.

ex) 
```
\nInvalid `photon.()` invocation in /app/server/graphql/resolvers/mutations.js:90:39\n\n\n\nReason: Error in connector: Record does not exist.\n
```

What I would expect in many cases is an error object that could allow me to decipher what happened through some properties.

```
error.message = 'Invalid `Photon.()` invocation. Reason: Record not found'
error.code = 'NOT_FOUND'
```

When this type of error ends up getting serialized say for a log, I don't have to parse through `\n` characters to get to the main intent.

I am trying to find the original errors I ran into when I first posted this issue. (I'll update once I find out what was causing it) In that case I was actually seeing unicode characters coming back in the errors that included coloring and formatting. Those were very hard to parse through and find what the actual error was if it didn't get printed to a terminal.

I think the pretty formatting is nice for local development but in production it makes it really hard to quickly get to the core issue.

**Comment by rizowski** [comment]
I updated my description with another example.

**Comment by schickling** [comment]
Thanks a lot for elaborating @rizowski. Your particular issue will be addressed by @divyenduz's current work on better error handling which will be implemented over the coming weeks.

TLDR: Every known error (such as "Record not found") will have a unique error code (incl. meta data) which you can use for custom error handling without the need to parse error message strings. 

I hope this addresses your problem in an even better way?

**Comment by divyenduz** [comment]
Update: Those are ANSI characters that makes Photon's output in the shell colorful etc. For now, you can pass Photon's error through a strip ANSI lib (https://www.npmjs.com/package/strip-ansi)

This spec (not implemented yet) talks about this more and would enable disabling it. https://github.com/prisma/specs/tree/master/photonjs#error-handling

**Comment by timsuchanek** [comment]
Thanks a lot for reporting 🙏
This issue is fixed in the latest alpha version of `prisma2`.
You can try it out with `npm i -g prisma2@alpha`.

In case it’s not fixed for you - please let us know and we’ll reopen this issue!

You can read here how to use this feature: https://github.com/prisma/specs/issues/377

**Comment by divyenduz** [comment]
Related issue: https://github.com/prisma/prisma2/issues/686
